Attempt at defining initialization, and simulation. (#3146)
modelica/ModelicaSpecification: chapters/introduction.tex

Therefore, a flat hybrid DAE is seen as a set of equations where some of the equations are only conditionally evaluated.
Initial setup of the model is specified using \lstinline!start!-attributes and equations that hold only during initialization.

> In terms of terminology the simulation starts with solving the initialization problem at the starting time, and then simulates forward in time from those computed initial values.

Reviewer: I'd prefer making proper definitions in the next section, so that they appear in the index, and that we don't put what is actually definitions right above the section named _Some Definitions_.

Author: Done!

Reviewer: To be continued here: https://github.com/modelica/ModelicaSpecification/pull/3146#discussion_r843502313

Resolving.

\end{definition}

\begin{definition}[Flattening]\index{flattening}

> The translation of a model described in Modelica to the corresponding model described as a hybrid DAE (see \cref{modelica-dae-representation}), involving expansion of inherited base classes, parameterization of base classes, local classes
> and components, and generation of connection equations from \lstinline!connect!-equations (basically, mapping the hierarchical structure of a model into a set of differential, algebraic and discrete equations together with the corresponding variable declarations and function definitions from the model).

Reviewer: Sentence-based line breaking:
```suggestion
The translation of a model described in Modelica to the corresponding model described as a hybrid DAE (see \cref{modelica-dae-representation}), involving expansion of inherited base classes, parameterization of base classes, local classes and components, and generation of connection equations from \lstinline!connect!-equations (basically, mapping the hierarchical structure of a model into a set of differential, algebraic and discrete equations together with the corresponding variable declarations and function definitions from the model).
```

Reviewer: Even better to break down into shorter sentences:
```suggestion
The translation of a model described in Modelica to the corresponding model described as a hybrid DAE (see \cref{modelica-dae-representation}), involving expansion of inherited base classes, parameterization of base classes, local classes and components, and generation of connection equations from \lstinline!connect!-equations.
In other words, mapping the hierarchical structure of a model into a set of differential, algebraic and discrete equations together with the corresponding variable declarations and function definitions from the model.
```

\end{definition}

\begin{definition}[Initialization]\index{initialization}
Simulation starts with solving the initialization problem at the starting time, resulting in values for all variables that are consistent with the result of the flattening.
\end{definition}

\begin{definition}[Transient analysis]\index{transient analysis}\index{initial value problem!see{transient analysis}}
Starting from the result of the initialization problem, the model is simulated forward in time.
This uses numerical methods for handling the hybrid DAE, resulting in solution trajectories for the model's variables, i.e., the value of the variables as a function of time.

> \end{definition}

Reviewer: This seems like the right place to have the document's only `\firstuse{initial value problem}`, allowing us to refer to this definition from the appendix if needed.

Author: I have it in the non-normative text that is now just below this; I think that suffices.

Reviewer: Agreed.  Resolving.

\begin{definition}[Simulation]\index{simulation}

> For a model simulation is the combination of the initialization followed by the transient analysis.

Reviewer: No need for qualification, and trying to be systematic about not overusing _the_ in front of these terms:
```suggestion
Simulation is the combination of initialization followed by transient analysis.
```

\end{definition}

\begin{definition}[Translation]\index{translation}
Translation is the process of preparing a Modelica simulation model for simulation, starting with flattening but not including the simulation itself.
\end{definition}

\begin{nonnormative}
The model can be analyzed in ways other than simulation, e.g., linearization, and parameter estimation, but they are not described in the specification.

> 

Reviewer: Maybe this could be moved to a new non-normative paragraph right below _simulation_? It would make it easier to find, and also make the remaining part of this non-normative section relate more closely to _translation_.

Author: Done.

In the numerical literature transient analysis is often called solving the \firstuse{initial value problem}, but that term is not used here to avoid confusion with the initialization problem.

> 

Reviewer: Maybe we should just move this non-normative part to a new non-normative part right below the definition of _transient analysis_, so that it isn't missed when reading the definition of _transient analysis_?

Author: Done.

and components, and generation of connection equations from \lstinline!connect!-equations (basically, mapping the hierarchical structure of a model into a set of differential, algebraic and discrete equations together with the corresponding variable declarations and function definitions from the model).
\end{definition}

> \begin{definition}[Simulation]\index{simulation}

Reviewer: Good start, but I think it would serve us well if we first had proper definitions of _initialization problem_ and (say) _transient analysis_, allowing _simulation_ to be defined as the combination of the two.

It would also give us a chance to move the `\firstuse{initial value problem}` to this introductory chapter, instead of this ending up in an appendix.

Then, we define _transient analysis_ (or whatever term we prefer) here, there are also some places affected by the PR where we should use this term instead of making awkward constructs based on _simulation_.

Author: Now done.

Reviewer: Nice.  I'll add a new comment above about the `\firstuse{initial value problem}`.  Resolving.

and components, and generation of connection equations from \lstinline!connect!-equations (basically, mapping the hierarchical structure of a model into a set of differential, algebraic and discrete equations together with the corresponding variable declarations and function definitions from the model).
\end{definition}

> \begin{definition}[Initialization]\index{initialization}

Reviewer: I suppose the term being defined actually includes _problem_?
```suggestion
\begin{definition}[Initialization problem]\index{def:initialization-problem}
```

Author: I prefer to define the current shorter 'Initialization'; to me the logic is that the Simulation starts with the Initialization which means _solving_ the Initialization problem. I'm not sure if that needs to be made clearer.

When we later use it we use terms like 'during initialization'.

> The simulation starts with solving the initialization problem at the starting time giving values for all variables that are consistent with the result of the flattening.

Reviewer: Some rewording and a comma:
```suggestion
Simulation starts with solving the initialization problem at the starting time, resulting in values for all variables that are consistent with the result of the flattening.
```

To me, "the result of the flattening" sounds a bit strange.  Wouldn't it be more important to stress the DAE aspect of the initialization problem than saying it comes out of flattening?

Author: I tried to keep the new terms to a minimum; so I prefer to keep 'flattening' here.

\end{definition}

\begin{definition}[Transient analysis]\index{transient analysis}\index{initial value problem!see{transient analysis}}

> Starting from the result of the initialization problem the model is then simulated forward in time.

Reviewer: Comma, skip "then":
```suggestion
Starting from the result of the initialization problem, the model is simulated forward in time.
```

> This generates values for the variables at different times -- and uses numerical methods for handling the hybrid DAE generated by the flattening.

Reviewer: Simpler to reorder, and rephrase "at different times"?
```suggestion
This uses numerical methods for handling the hybrid DAE, resulting in solution trajectories over time for the model's variables.
```

Author: Something like that - but trajectories over time seems odd so I will reformulate.

> \end{definition}

Reviewer: How about a cross reference to `\cref{modelica-dae-representation}` here?

Author: Yes, I'll add a cross-reference a few lines above.

Reviewer: OK.  Resolving.

\begin{definition}[Simulation]\index{simulation}

> For a model simulation is the combination of the initialization followed by the transient analysis.

Reviewer: I suppose we could let the condition on applying to a simulation model be implicit:
```suggestion
Simulation is the combination of solving the initialization problem and carrying out transient analysis.
```

Author: I will apply the first change, but as previously noted defining 'Initialization' means that I avoid the other changes.

Reviewer: Sure, let's try to keep down use of _initialization problem_.  Resolving.

\end{definition}

> \begin{definition}[Translation]\index{translation}

Reviewer: Would it be better to place this definition right after _flattening_?

Author: That would require a forward reference between the definitions (even if simulation is excluded it is still mentioned), and I wanted to avoid that.

Reviewer: Sure, it's a tradeoff between avoiding forward references and giving the definitions in the natural chronological order.  Resolving.

> The translation starts with the flattening and includes everything needed for the simulation of the model, but not the simulation itself.

Reviewer: Attempting a somewhat more concrete way of saying the same thing:
```suggestion
Translation is the process of preparing a Modelica simulation model for simulation, starting with flattening but not including the simulation itself.
```
